packages/text-annotator/src/SelectionHandler.ts

```diff
@@ -317,6 +305,29 @@ export const SelectionHandler = (
 
   hotkeys(ARROW_KEYS.join(','), { keydown: true, keyup: false }, handleArrowKeyPress);
 
+  // Helper
+  const upsertCurrentTarget = () => {
+    const existingAnnotation = store.getAnnotation(currentTarget.annotation);
+    if (!existingAnnotation) {
+      store.addAnnotation({
+        id: currentTarget.annotation,
+        bodies: [],
+        target: currentTarget
+      });
+      return;
+    }
+
+    const { target: { updated: existingTargetUpdated } } = existingAnnotation;
+    const { updated: currentTargetUpdated } = currentTarget;
+    if (
+      !existingAnnotation ||
```

Comment:
I think this one is redundant?

Reply:
Hmm, I'm not sure that it's "redundant". As the `update` call was extracted into the `upsert` in the selection revision PR, https://github.com/recogito/text-annotator-js/pull/151, [source](https://github.com/recogito/text-annotator-js/pull/151/files#diff-1e072053802159d91da89e331e1fe166da3cb395a3fc6ab3d6317b0510ea17e0R174-R186). So there might was a reason for its existence 🤔

Reply:
And if the `update` is needed in the `upsert` - comparing the `updated` timestamp is the key to preventing redundant calls to the `store.observe`

Reply:
Sorry, that wasn't clear from my side. I meant just the '!existingAnnotation' specifically. existingAnnotation is always true at this point unless I'm missing something.

Reply:
Right... 
From the runtime perspective the value should exist at the point. By TS doesn't know about it due to the `updated?: Date` definition. So if we would have the strict null check enabled - it would complain that `existingTargetUpdated < currentTargetUpdated` would run on possibly `undefined` value. So I added the `!existingAnnotation` out of habbit

Reply:
OHHHHH! I was meant to be the: 
```suggestion
      !existingTargetUpdated ||
```

Reply:
Glad I asked :-)